utils/validate_pr_quickstarts.js

```diff
@@ -0,0 +1,359 @@
+'use strict';
+
+const glob = require('glob');
+
+const {
+  fetchPaginatedGHResults,
+  filterOutTestFiles,
+} = require('./github-api-helpers');
+const {
+  findMainQuickstartConfigFiles,
+  readQuickstartFile,
+  removeRepoPathPrefix,
+  passedProcessArguments,
+} = require('./helpers');
+const {
+  fetchNRGraphqlResults,
+  translateMutationErrors,
+} = require('./nr-graphql-helpers');
+
+const GITHUB_API_URL = passedProcessArguments[0];
+const GITHUB_REPO_BASE_URL =
+  'https://github.com/newrelic/newrelic-quickstarts/tree/main';
+const GITHUB_RAW_BASE_URL =
```

Comment:
In the case of a brand new quickstart, URLs build with either the `GITHUB_REPO_BASE_URL` or `GITHUB_RAW_BASE_URL` bases would not exist on the  `main` branch of the repo.

For values on the mutation input object referencing those constants —like the `icon`, `sourceUrl`, or screenshot `url` attributes—would we need to construct URLs based on the PR or a given file's most recent commit?

Something like: https://raw.githubusercontent.com/tyreer/newrelic-quickstarts/80857af8a45a481d144b68bd8206a8d7d804a220/quickstarts/rob-test-qs-invalid/logo.png (from [here](https://github.com/tyreer/newrelic-quickstarts/blob/80857af8a45a481d144b68bd8206a8d7d804a220/quickstarts/rob-test-qs-invalid/logo.png
))

Question maybe is: Is the catalog service reading the contents at that URL and performing validation logic, or simply confirming this is a valid URL?

Reply:
Not seeing any objections, so I'm going to stick with the `/main/` path